### DISCUSSION TITLE
Issue with `T.ToUndirected()` and heterogeneous data

### DISCUSSION THREAD

**Discussion by maximocrv** [pyg-team/pytorch_geometric#3261] [Q&A]
I have followed the guide provided [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html#) with a custom dataset, and when I try to call `T.ToUndirected()(custom_data)` I get the following error: `RuntimeError: index -1 is out of bounds for dimension 0 with size 80083`, which arises from the call to `out.scatter_add_` in `torch_scatter/scatter.py`. On the other hand `T.AddSelfLoops()` and `T.NormalizeFeatures()` seem to work fine, so I am not sure what the issue is. Any ideas as to what the problem could be?

**Comment by wsad1** [comment]
@maximocrv could you please your data or a sample of data that captures the traits of your data well. That would help debug this faster.

**Reply by maximocrv** [reply]
Hi, I updated PyG to the latest commit and without making any changes to the code it now works fine, perhaps there was some bug previously

**Reply by rusty1s** [reply]
Thanks for reporting. There was a bug in `ToUndirected` in case a graph has the same number of nodes and edges. This probably introduced your issues.

**Reply by shahinghasemi** [reply]
> Hi, I updated PyG to the latest commit and without making any changes to the code it now works fine, perhaps there was some bug previously

How did you update to the latest commit? I've installed PyG using conda and I have the same problem however don't know how to update to the latest commit? @rusty1s would you please help. I'm blocked.

**Reply by rusty1s** [reply]
This is only possible when installing via `pip`:
```
pip install git+https://github.com/pyg-team/pytorch_geometric.git
```

**Reply by shahinghasemi** [reply]
Thanks for the replay @rusty1s , I first removed the old one (`conda remove pyg`) and then executed the above command however now `import torch_geometric.nn` can't be resolved in my conda environment. What am I missing?
An additional question: What's the reason that you don't update the package semantically? This way it's far convenient to update it using `conda update package` to use the latest version/commit of it.

**Reply by rusty1s** [reply]
What do you mean with updating the package semantically? We continuously release new versions with new features and bugfixes. I'll probably release `2.0.3` in the upcoming week.

Can you check that you are indeed installing into the conda directory? What's the output of:
```
which python
which pip
```

**Reply by shahinghasemi** [reply]
>What do you mean with updating the package semantically? We continuously release new versions with new features and bugfixes. I'll probably release 2.0.3 in the upcoming week.

I mean updating the package more frequently. for example (if I'm not wrong) `RandomLinkSplit` or `ToUndirected` is not updated to the latest commit which are important functions though. 

>Can you check that you are indeed installing into the conda directory? What's the output of:

which python: /home/shahin/miniconda3/envs/hetero/bin/python
which pip: /home/shahin/miniconda3/envs/hetero/bin/pip

**Reply by rusty1s** [reply]
I have plans to release `2.0.3` tomorrow. Please stay tuned :)
